pkg/manifests/manifests.go

```diff
@@ -843,9 +858,72 @@ func (f *Factory) PrometheusK8s(host string) (*monv1.Prometheus, error) {
 		}
 	}
 
+	if trustedCABundleCM != nil {
+		volumeName := "prometheus-trusted-ca-bundle"
+		p.Spec.Containers = append(p.Spec.Containers, v1.Container{
+			Name: "prometheus",
+			VolumeMounts: []v1.VolumeMount{
+				trustedCABundleVolumeMount(volumeName, PrometheusTrustedCABundleDir),
+			},
+		})
+		p.Spec.Volumes = append(p.Spec.Volumes, trustedCABundleVolume(trustedCABundleCM.Name, volumeName))
+	}
+
+	telemetryEnabled := f.config.TelemeterClientConfig.IsEnabled()
+	if telemetryEnabled {
+
+		selectorRelabelConfig, err := relabelgen.LabelSelectorsToRelabelConfig(f.config.PrometheusK8sConfig.TelemetryMatches)
+		if err != nil {
+			return nil, errors.Wrap(err, "generate label selector relabel config")
+		}
+
+		compositeToken, err := json.Marshal(map[string]string{
+			"cluster_id":          f.config.TelemeterClientConfig.ClusterID,
+			"authorization_token": f.config.TelemeterClientConfig.Token,
+		})
+
+		spec := monv1.RemoteWriteSpec{
+			URL:         f.config.TelemeterClientConfig.TelemeterServerURL,
+			BearerToken: base64.StdEncoding.EncodeToString(compositeToken),
+			WriteRelabelConfigs: []monv1.RelabelConfig{
+				*selectorRelabelConfig,
+				monv1.RelabelConfig{
+					TargetLabel: "_id",
+					Replacement: f.config.TelemeterClientConfig.ClusterID,
+				},
+				monv1.RelabelConfig{
+					SourceLabels: []string{"__name__"},
+					TargetLabel:  "__name__",
+					Regex:        "ALERTS",
+					Replacement:  "alerts",
+				},
+			},
+		}
+
+		if trustedCABundleCM != nil {
+			spec.TLSConfig = &monv1.TLSConfig{
+				CAFile: PrometheusTrustedCABundlePath,
+			}
+		}
+
+		p.Spec.RemoteWrite = []monv1.RemoteWriteSpec{spec}
+
+	}
+	if !telemetryEnabled {
+		p.Spec.RemoteWrite = nil
+	}
+
 	if len(f.config.PrometheusK8sConfig.RemoteWrite) > 0 {
 		p.Spec.RemoteWrite = f.config.PrometheusK8sConfig.RemoteWrite
 	}
+	for _, rw := range p.Spec.RemoteWrite {
+		if f.config.HTTPConfig.HTTPProxy != "" {
+			rw.ProxyURL = f.config.HTTPConfig.HTTPProxy
+		}
+		if f.config.HTTPConfig.HTTPSProxy != "" {
+			rw.ProxyURL = f.config.HTTPConfig.HTTPSProxy
+		}
```

Comment:
afaik the proxy config also has to respect the trusted CA bundle and react to it upon changes. Is this something we can add easily with the existing Prometheus infrastructure without restarting the whole pod?

Reply:
clarified in https://github.com/openshift/cluster-monitoring-operator/pull/434#discussion_r331925798